### PR TITLE
Remove the kBraveNewsFeature Flag

### PR DESCRIPTION
**Pull request by fallaciousreasoning**
Resolves https://github.com/brave/brave-browser/issues/30576

## Submitter Checklist:

- [ ] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: browser/browsing_data/brave_browsing_data_remover_delegate.cc [hunk 1]
```diff
@@ -66,12 +66,10 @@ void BraveBrowsingDataRemoverDelegate::RemoveEmbedderData(
   if (remove_mask & content::BrowsingDataRemover::DATA_TYPE_CACHE)
     ClearIPFSCache();
 #endif
-  if (base::FeatureList::IsEnabled(brave_news::features::kBraveNewsFeature)) {
```
Comment (iefremov) [inline browser/browsing_data/brave_browsing_data_remover_delegate.cc:69]: pls make sure the corresponding `#include`s are removed as well in all files

File: components/brave_ads/browser/ads_service_impl.cc [hunk 2]
```diff
@@ -354,9 +354,7 @@ void AdsServiceImpl::OnGetDeviceId(std::string device_id) {
 }
 
 bool AdsServiceImpl::UserHasOptedIn() const {
-  return base::FeatureList::IsEnabled(
-             brave_news::features::kBraveNewsFeature) &&
```
Comment (tmancey) [inline components/brave_ads/browser/ads_service_impl.cc:358]: can we please remove `#include "brave/components/brave_news/common/features.h"` from `ads_service_impl.cc`, thanks

File: browser/browsing_data/brave_browsing_data_remover_delegate.cc (hunk 1 above)
Comment (fallaciousreasoning) [reply browser/browsing_data/brave_browsing_data_remover_delegate.cc:69]: Good call!

**Review by SergeyZhukovsky — APPROVED** [review]
++

File: chromium_src/chrome/browser/flags/android/chrome_feature_list.cc [hunk 3]
```diff
@@ -23,7 +23,6 @@
 #define kForceWebContentsDarkMode kForceWebContentsDarkMode,            \
     &brave_rewards::features::kBraveRewards,                            \
     &brave_search_conversion::features::kOmniboxBanner,                 \
-    &brave_news::features::kBraveNewsFeature,                           \
     &brave_news::features::kBraveNewsV2Feature,                         \
```
Comment (tmancey) [inline chromium_src/chrome/browser/flags/android/chrome_feature_list.cc:26]: LGTM (do we not need to check for the new Brave News V2 feature whether it is enabled or not?)

Comment (fallaciousreasoning) [reply chromium_src/chrome/browser/flags/android/chrome_feature_list.cc:26]: Nope, the Brave News V2 feature is a UI refresh - I'm removing that flag too https://github.com/brave/brave-core/pull/18602

**Comment by fallaciousreasoning** [comment]
@iefremov would you mind taking another look?

**Review by mkarolin — APPROVED** [review]
`chromium_src`++

File: browser/profiles/brave_profile_manager.cc [hunk 4]
```diff
@@ -117,9 +116,7 @@ void BraveProfileManager::DoFinalInitForServices(Profile* profile,
   DCHECK(status);
   status->UpdateGCMDriverStatus();
 #endif
-  if (base::FeatureList::IsEnabled(brave_news::features::kBraveNewsFeature)) {
-    brave_news::BraveNewsControllerFactory::GetForContext(profile);
-  }
+  brave_news::BraveNewsControllerFactory::GetForContext(profile);
```
Comment (bridiver) [inline browser/profiles/brave_profile_manager.cc:119]: this is not correct, you should return true from `BrowserContextKeyedServiceFactory::ServiceIsCreatedWithContext`

Comment (fallaciousreasoning) [reply browser/profiles/brave_profile_manager.cc:119]: Do you mean this should override `ServiceIsCreatedWithContext` and we should call `brave_news::BraveNewsControllerFactory::GetInstance()` here instead?

Is it just that its more idiomatic, or will there be some behavioral change?

I did notice that there seem to be two methods `GetForContext` and `GetControllerForContext` which do the same thing here.

Comment (fallaciousreasoning) [reply browser/profiles/brave_profile_manager.cc:119]: Ah, I see - we already call `GetInstance` in `EnsureBrowserContextKeyedServiceFactoriesBuilt`

Comment (fallaciousreasoning) [reply browser/profiles/brave_profile_manager.cc:119]: https://github.com/brave/brave-core/pull/24718

**Pull request merged**